### ISSUE TITLE
WebView appears as white until all Web Components have loaded

### ISSUE THREAD

**Issue by pre** [AppGyver/steroids#830]
WebView is white until ALL web components have fully loaded. This is different behavior than what a normal page load looks like without using web components. Super-navbar etc are implemented with polyfills. Problem is most apparent on older (slower) Android devices where web components do not load as fast as they do on newer devices. 

An aggregate of the problem is that Loading.html is dimissed when the WebView is ready, although the content of the WebView appear as white until ALL web components are ready.

**Comment by AntonVoltchok** [comment]
Hi , just wanted to follow up to see if you figured out a solution, setting a delay on dismissing loading.HTML might be a possibility but there's got to be a more practical way.

**Comment by mikestokes** [comment]
+1

**Comment by kaipk** [comment]
+1 same problem with crosswalk. sometimes the black screen is persistent in the view.

**Comment by justincolangelo** [comment]
Calling `supersonic.ui.initialView.show();` from the main view (in my case logging out of the application and showing the log in view) shows the initial view, but then flashes the black screen until the screen is touched. This is on multiple Android phones.

**Comment by harsaharsa** [comment]
@justincolangelo are you running any JS code when you switch back to the view, i.e. with a visibilitychange event?

**Comment by justincolangelo** [comment]
@harsaharsa, Yes. I have a controller attached to the initialView that has a `supersonic.ui.views.current.whenVisible( function(){});` method running.

**Comment by dantabel** [comment]
+1

**Comment by Blechfalke** [comment]
+1

**Comment by alexkuttig** [comment]
+1

**Comment by poppahorse** [comment]
+1

**Comment by alexkuttig** [comment]
+1

**Comment by gkachru** [comment]
+1

**Comment by jeffthompson1971** [comment]
I'm seeing this even today.   Is there a fix for this?  nearly a year old it looks like...

**Comment by poppahorse** [comment]
Nope :(

Also @pre this is happening for me on a Nexus 6P, so the old / slow device comment isn't true

Its little things like this that really prevent my steroids apps feeling really polished and tight

**Comment by harsaharsa** [comment]
It's a very fundamental problem and not easy to solve – we haven't found a solution ourselves yet. If you don't utilize Web Components but instead initiate everything through the JS APIs, the situation should be better.

**Comment by gkachru** [comment]
A workaround is to create one's own loading screen and manually load views. Pre-loading the loading view makes it work quite well visually, even on low end devices, but costs memory. So your touch handler for drawer items may look something like:

```
function LoadView(view_id, view_location) {
    supersonic.ui.drawers.close('left');
    supersonic.ui.layers.replace('loader#loading');
    CustomViewLoader(view_id, view_location).then(function(view_instance) {
        supersonic.ui.layers.replace(view_instance);
    });
}
```
